src/test/rgw/test_ratelimit_unittest.cc

```diff
@@ -0,0 +1,191 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include <gtest/gtest.h>
+#include "rgw/rgw_qos.h"
+
+
+using namespace std::chrono_literals;
+
+TEST(RGWRateLimit, shouldratelimit1)
+{
+  // info.enabled = false, so no limit
+  std::unique_ptr<CephContext> cct = std::make_unique<CephContext>(CEPH_ENTITY_TYPE_ANY);
```

Comment:
tests shouldn't need to create their own `CephContext`. you can just add ` $<TARGET_OBJECTS:unit-main>` to the executable for access to `g_ceph_context`. same for test_ratelimit.cc and test_ratelimit_gc.cc

you can find examples of this in src/test/rgw/CMakeLists.txt, like `unittest_rgw_compression`. the `unit-main` target just adds `src/test/unit.cc`, which has a `main()` function that calls `global_init()` and `common_init()` to initialize `g_ceph_context`

Reply:
It looks like the main function on the bench_* files is duplicated with the one from src/test/unit.cc